File: nordeus_challenge_backend_spring_boot/src/main/java/com/mmileticc/nordeuschallenge/model/dto/BattleTurnResultDTO.java
package com.mmileticc.nordeuschallenge.model.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BattleTurnResultDTO {

    private Long heroId;
    private Long monsterId;

    private int heroCurrentHealth;
    private int heroAttack;
    private int heroDefense;
    private int heroMagic;

    private int monsterCurrentHealth;
    private int monsterAttack;
    private int monsterDefense;
    private int monsterMagic;

    private String heroMoveName;
    private String monsterMoveName;

    private boolean battleFinished;
    private String winner;

    private List<String> logs = new ArrayList<>();
    private List<ActiveBuffDTO> activeBuffs = new ArrayList<>();
}

File: Gauntlet of the Hero - GOTH/Assets/_Scripts/Models/Backend/BattleTurnResultDto.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace GauntletOfTheHero.Backend.Models
{
    [System.Serializable]
    public class BattleTurnResultDto
    {
        [JsonProperty("heroId")]
        public long heroId;

        [JsonProperty("monsterId")]
        public long monsterId;

        [JsonProperty("heroCurrentHealth")]
        public int heroCurrentHealth;

        [JsonProperty("heroAttack")]
        public int heroAttack;

        [JsonProperty("heroDefense")]
        public int heroDefense;

        [JsonProperty("heroMagic")]
        public int heroMagic;

        [JsonProperty("monsterCurrentHealth")]
        public int monsterCurrentHealth;

        [JsonProperty("monsterAttack")]
        public int monsterAttack;

        [JsonProperty("monsterDefense")]
        public int monsterDefense;

        [JsonProperty("monsterMagic")]
        public int monsterMagic;

        [JsonProperty("heroMoveName")]
        public string heroMoveName;

        [JsonProperty("monsterMoveName")]
        public string monsterMoveName;

        [JsonProperty("battleFinished")]
        public bool battleFinished;

        [JsonProperty("winner")]
        public string winner;

        [JsonProperty("logs")]
        public List<string> logs = new List<string>();

        [JsonProperty("activeBuffs")]
        public List<ActiveBuffDto> activeBuffs = new List<ActiveBuffDto>();
    }
}